tests/contenttypes_tests/test_fields.py

```diff
@@ -27,9 +27,28 @@ def test_incorrect_get_prefetch_queryset_arguments(self):
 
 class GenericRelationTests(TestCase):
 
+    def setUp(self):
+        self.question = Question.objects.create(text='test')
+        self.answer1 = Answer.objects.create(question=self.question)
+        self.answer2 = Answer.objects.create(question=self.question)
+
     def test_value_to_string(self):
-        question = Question.objects.create(text='test')
-        answer1 = Answer.objects.create(question=question)
-        answer2 = Answer.objects.create(question=question)
-        result = json.loads(Question.answer_set.field.value_to_string(question))
-        self.assertCountEqual(result, [answer1.pk, answer2.pk])
+        result = json.loads(Question.answer_set.field.value_to_string(self.question))
+        self.assertCountEqual(result, [self.answer1.pk, self.answer2.pk])
+
+    def test_remove_deletes_related_object_from_database(self):
+        self.assertTrue(self.answer1 in Answer.objects.all())
+        self.assertTrue(self.answer2 in Answer.objects.all())
+
+        self.question.answer_set.remove(self.answer1)
+
+        self.assertFalse(self.answer1 in Answer.objects.all())
+        self.assertTrue(self.answer2 in Answer.objects.all())
+        self.assertTrue(self.question in Question.objects.all())
+
+    def test_clear_deletes_all_related_objects_from_database(self):
+        self.question.answer_set.clear()
+
+        self.assertFalse(self.answer1 in Answer.objects.all())
+        self.assertFalse(self.answer2 in Answer.objects.all())
+        self.assertTrue(self.question in Question.objects.all())
```

Comment:
I would move new tests to a `generic_relations.tests.GenericRelationsTests`, e.g.
```diff
diff --git a/tests/generic_relations/tests.py b/tests/generic_relations/tests.py
index 0eb9dee03b..002dab20e0 100644
--- a/tests/generic_relations/tests.py
+++ b/tests/generic_relations/tests.py
@@ -25,10 +25,10 @@ class GenericRelationsTests(TestCase):
         cls.quartz = Mineral.objects.create(name='Quartz', hardness=7)
 
         # Tagging stuff.
-        cls.bacon.tags.create(tag='fatty')
-        cls.bacon.tags.create(tag='salty')
-        cls.lion.tags.create(tag='yellow')
-        cls.lion.tags.create(tag='hairy')
+        cls.fatty = cls.bacon.tags.create(tag='fatty')
+        cls.salty = cls.bacon.tags.create(tag='salty')
+        cls.yellow = cls.lion.tags.create(tag='yellow')
+        cls.hairy = cls.lion.tags.create(tag='hairy')
 
     def comp_func(self, obj):
         # Original list of tags:
@@ -343,6 +343,30 @@ class GenericRelationsTests(TestCase):
         self.assertEqual(1, bacon.tags.count())
         self.assertEqual(1, qs.count())
 
+    def test_clear(self):
+        self.assertSequenceEqual(
+            TaggedItem.objects.order_by('tag'),
+            [self.fatty, self.hairy, self.salty, self.yellow],
+        )
+        self.bacon.tags.clear()
+        self.assertSequenceEqual(self.bacon.tags.all(), [])
+        self.assertSequenceEqual(
+            TaggedItem.objects.order_by('tag'),
+            [self.hairy, self.yellow],
+        )
+
+    def test_remove(self):
+        self.assertSequenceEqual(
+            TaggedItem.objects.order_by('tag'),
+            [self.fatty, self.hairy, self.salty, self.yellow],
+        )
+        self.bacon.tags.remove(self.fatty)
+        self.assertSequenceEqual(self.bacon.tags.all(), [self.salty])
+        self.assertSequenceEqual(
+            TaggedItem.objects.order_by('tag'),
+            [self.hairy, self.salty, self.yellow],
+        )
+
     def test_generic_relation_related_name_default(self):
         # GenericRelation isn't usable from the reverse side by default.
         msg = (
```